core/src/main/scala/dagr/core/tasksystem/Task.scala

```diff
@@ -23,20 +23,167 @@
  */
 package dagr.core.tasksystem
 
-import com.fulcrumgenomics.commons.CommonsDef.unreachable
+import java.time.{Duration, Instant}
+
+import com.fulcrumgenomics.commons.CommonsDef.{FilePath, unreachable}
+import com.fulcrumgenomics.commons.io.PathUtil
+import com.fulcrumgenomics.commons.util.TimeUtil.formatElapsedTime
+import dagr.api.models.{ResourceSet, TaskStatus, TimePoint}
+import dagr.core.DagrDef.TaskId
+import dagr.core.exec.Executor
 import dagr.core.execsystem.TaskExecutionInfo
+import dagr.core.tasksystem.Task.TaskInfo
 
 import scala.collection.mutable
 import scala.collection.mutable.ListBuffer
+import scala.reflect.ClassTag
 import scala.util.control.Breaks._
 
 /** Utility methods to aid in working with a task. */
 object Task {
 
+  trait TaskInfoLike extends dagr.api.models.TaskInfoLike with Ordered[TaskInfoLike] {
+    /** Gets the associated task. */
+    def task: Task
+
+    def compare(that: TaskInfoLike): Int = {
+      (this.id, that.id) match {
+        case (Some(_), None)    => -1
+        case (None, Some(_))    => 1
+        case (None, None)       => this.status.ordinal - that.status.ordinal
+        case (Some(l), Some(r)) => (l - r).toInt
+      }
+    }
+
+    def scriptPath: Option[FilePath] = this.script.map(PathUtil.pathTo(_))
+    def logPath: Option[FilePath]    = this.script.map(PathUtil.pathTo(_))
+
+
+  }
+
+  /** Execution information associated with a task.  Any execution system should extend this class to store
+    * their specific metadata.
+    * @param task the task in question
+    * @param initStatus the initial status
+    * @param id the unique id, if any, of the task.  The id may not be set until execution in some execution systems.
+    * @param attempts the number of execution attempts (one-based)
+    * @param script the path to the execution script for the task, if any
+    * @param log the path to the log file for the task, if any
+    * @param resources the resources that the tasks used during execution or was scheduled with
+    * @param exitCode the exit-code for the task, if any
+    * @param throwable a throwable generated during execution, if any
+    */
+  private[dagr] abstract class TaskInfo
```

Comment:
The `TaskInfo` for the execution systems.  Updated so we can have various callbacks handled internally.